pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -2209,5 +2234,11 @@ boolean isErrorStat() {
         return errorState;
     }
 
+    @VisibleForTesting
```

Comment:
Unless I'm missing something, I think we can get rid of this annotation since it is a part of the public API, and it is visible for more than just testing.

Reply:
Makes sense to me. I agree it makes sense besides testing.